### PR TITLE
Add delimiters in YAML format

### PR DESCRIPTION
**Pull request by Seopril**
This is an intended fix for issue #1065. The yaml export now functions as shown in the "Expected Behavior" described in that bug report. 

### Example Output:

```
---
title: Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
date: 2021-01-05 11:26
starred: False
tags: 
...

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

### Checklist

- [X] I have read the [contributing doc](https://github.com/jrnl-org/jrnl/blob/develop/CONTRIBUTING.md).
- [X] I have included a link to the relevant issue number.
- [X] I have tested this code locally.
- [X] I have checked to ensure there aren't other open [pull requests](../pulls)
  for the same issue.
- [X] I have written new tests for these changes, as needed.
- [X] All tests pass.

### CODE REVIEW

File: features/format.feature [hunk 1]
```diff
@@ -410,10 +410,12 @@ Feature: Custom formats
         """
         And the content of file "2020-08-29_entry-the-first.md" in the cache should be
         """
+        ---
```
Comment (wren) [inline features/format.feature:413]: I think the example in #1065 meant the dashes here as a placeholder for content earlier in the file (not actual dashes).

File: features/format.feature [hunk 2]
```diff
@@ -410,10 +410,12 @@ Feature: Custom formats
...
         title: Entry the first.
         date: 2020-08-29 11:11
         starred: False
         tags: tagone, ipsum, tagtwo
+        ...
```
Comment (wren) [inline features/format.feature:437]: I think the example in #1065 meant the dots here as a placeholder for content later in the file (not actual dots).

File: features/format.feature [hunk 3]
```diff
@@ -458,11 +460,13 @@ Feature: Custom formats
         """
         And the content of file "2020-09-24_the-third-entry-finally-after-weeks-without-writing.md" in the cache should be
         """
+        ---
         title: The third entry finally after weeks without writing.
         date: 2020-09-24 09:14
         starred: False
         tags: tagone, tagthree
```
Comment (wren) [inline features/format.feature:467]: This is missing the `body: |` part of the description in #1065. This is the main part of the yaml exporter than needs fixing. The `body` key needs to be output (followed by `: |`), then the body of the entry should follow on the next line, and be indented evenly, and indented farther than the `body` key.

**Review by wren — CHANGES_REQUESTED** [review]
Thank you for taking this one on!

I think there was a little confusion with the description in the ticket, so I left some clarifying comments inline.

File: features/format.feature (hunk 1 above)
Comment (wren) [reply features/format.feature:413]: Oh, hey, these dashes actually part of the yaml spec. My mistake!

https://yaml.org/spec/1.2/spec.html#id2760395

File: features/format.feature (hunk 2 above)
Comment (wren) [reply features/format.feature:437]: Oh, hey, these dots actually part of the yaml spec. My mistake!

https://yaml.org/spec/1.2/spec.html#id2760395

File: features/format.feature (hunk 1 above)
Comment (Seopril) [reply features/format.feature:413]: I think the example was supposed to be actual dashes and dots based on Pandoc documentation that was linked in the ticket. https://pandoc.org/MANUAL.html#metadata-blocks

"A YAML metadata block is a valid YAML object, delimited by a line of three hyphens (---) at the top and a line of three hyphens (---) or three dots (...) at the bottom."

Didn't refresh and just saw your second comment. My bad

**Comment by Seopril** [comment]
I've added in the body section for the YAML export and updated the feature tests accordingly.

**Review by wren — APPROVED** [review]
Thank you!

**Pull request merged**